test/unit/mbyte_spec.lua

```diff
@@ -347,4 +347,10 @@ describe('mbyte', function()
     check('L̓̉̑̒̌̚ơ̗̌̒̄̀ŕ̈̈̎̐̕è̇̅̄̄̐m̖̟̟̅̄̚', {'L̓̉̑̒̌̚', 'ơ̗̌̒̄̀', 'ŕ̈̈̎̐̕', 'è̇̅̄̄̐', 'm̖̟̟̅̄̚'})
     -- stylua: ignore end
   end)
+
+  itp("utf_fold doesn't crash on invalid codepoints", function()
+    eq(55296, lib.utf_fold(55296)) -- 0xd800 (high surrogate, invalid as a character)
```

Comment:
I was going to say that this may be indicating a problem somewhere else in Neovim (why would we have invalid characters here when my backtrace shows valid ascii characters?) but actually, printing the `regprog` of my reproducer, I see:
```
pattern = 0x55f27cc8e130 "#[A-Za-z\\u00ac-\\uffff_@][A-Za-z\\u00ac-\\uffff0-9_@-]*",
```
Which comes from [vim-polyglot's css syntax file](https://github.com/sheerun/vim-polyglot/blob/master/syntax/css.vim#L70). It's completely normal to encounter invalid characters when iterating over a range of characters, so I guess everything's fine!